src/benchmark/operators/join_aggregate_benchmark.cpp

```diff
@@ -0,0 +1,175 @@
+#include <memory>
+#include <numeric>
+#include <vector>
+
+#include "../micro_benchmark_basic_fixture.hpp"
+#include "benchmark/benchmark.h"
+#include "expression/expression_functional.hpp"
+#include "operators/aggregate_hash.hpp"
+#include "operators/aggregate_sort.hpp"
+#include "operators/join_hash.hpp"
+#include "operators/join_sort_merge.hpp"
+#include "operators/table_wrapper.hpp"
+#include "storage/table.hpp"
+#include "storage/value_segment.hpp"
+#include "types.hpp"
+
+namespace {
+
+constexpr auto TABLE_SIZE = size_t{1'000};
+constexpr auto NUMBER_OF_CHUNKS_JOIN_AGGREGATE = size_t{1};
+// How much of the table size range should be used in the join columns.
+// The lower the selectivity, the higher the collision rate is and the more
+// values are equal in the join columns
+constexpr auto SELECTIVITY = 0.2;
+
+}  // namespace
+
+namespace opossum {
+
+using namespace opossum::expression_functional;  //NOLINT
+
+pmr_vector<int32_t> generate_ids(const size_t table_size) {
+  auto values = pmr_vector<int32_t>(table_size);
+  const auto range = static_cast<int>(TABLE_SIZE * SELECTIVITY);
+
+  // The result ids are always the same in each table because of the seed
+  unsigned int seed = 54321;
+  for (size_t row_index = 0; row_index < table_size; ++row_index) {
+    values[row_index] = rand_r(&seed) % range + 1;
```

Comment:
Please avoid using `rand_r()` and similar function. Better use their modern equivalents in C++, e.g., `std::random_device`.

Reply:
Replaced in 32b66c6

Reply:
Why use such a special engine (can't say that I understand what it does) instead of `default_random_engine`?

Reply:
I changed it be use `default_random_engine`